django/core/validators.py

```diff
@@ -163,29 +163,27 @@ def validate_integer(value):
 
 @deconstructible
 class EmailValidator:
+    # Matches https://html.spec.whatwg.org/multipage/forms.html#valid-e-mail-address
     message = _('Enter a valid email address.')
     code = 'invalid'
     user_regex = _lazy_re_compile(
-        r"(^[-!#$%&'*+/=?^_`{}|~0-9A-Z]+(\.[-!#$%&'*+/=?^_`{}|~0-9A-Z]+)*\Z"  # dot-atom
-        r'|^"([\001-\010\013\014\016-\037!#-\[\]-\177]|\\[\001-\011\013\014\016-\177])*"\Z)',  # quoted-string
+        r"^[a-z0-9.!#$%&'*+\/=?^_`{|}~-]+\Z",
         re.IGNORECASE)
     domain_regex = _lazy_re_compile(
-        # max length for domain name labels is 63 characters per RFC 1034
-        r'((?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+)(?:[A-Z0-9-]{2,63}(?<!-))\Z',
+        r'[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?(?:\.[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?)*\Z',
         re.IGNORECASE)
     literal_regex = _lazy_re_compile(
         # literal form, ipv4 or ipv6 address (SMTP 4.1.3)
         r'\[([A-f0-9:\.]+)\]\Z',
         re.IGNORECASE)
-    domain_whitelist = ['localhost']
 
     def __init__(self, message=None, code=None, whitelist=None):
         if message is not None:
             self.message = message
         if code is not None:
             self.code = code
         if whitelist is not None:
-            self.domain_whitelist = whitelist
+            pass  # deprecated
```

Comment:
According to https://html.spec.whatwg.org/multipage/forms.html#valid-e-mail-address, the text after `@` can be a letter, digit or hyphen optionally followed by a `.` and a trail of letters, digits or hyphens. 

The domain whitelist was there to allow domains that don't have a `.` in them to be treated as a special case (and be allowed. However, since the validation now considers a `.` as optional already, we don't have to specify any whitelists manually.